network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/AbstractLoadingLimits.java

```diff
@@ -73,8 +73,13 @@ public T setPermanentLimit(double permanentLimit) {
 
     @Override
     public Collection<LoadingLimits.TemporaryLimit> getTemporaryLimits() {
-        return attributes.getTemporaryLimits() == null ? Collections.emptyList()
-                                                       : attributes.getTemporaryLimits().values().stream().map(TemporaryLimitImpl::new).collect(Collectors.toUnmodifiableList());
+        if (attributes.getTemporaryLimits() == null) {
+            return Collections.emptyList();
+        }
+        // make sure limits are provided by descending durations
+        TreeMap<Integer, TemporaryLimitAttributes> sortedLimitAttributes = new TreeMap<>(ACCEPTABLE_DURATION_COMPARATOR);
+        sortedLimitAttributes.putAll(attributes.getTemporaryLimits());
+        return sortedLimitAttributes.values().stream().map(TemporaryLimitImpl::new).collect(Collectors.toUnmodifiableList());
```

Comment:
attributes.getTemporaryLimits().entrySet().stream().sorted(Map.Entry.comparingByKey(ACCEPTABLE_DURATION_COMPARATOR)).map(Map.Entry::getValue).collect(Collectors.toList());

Reply:
yes I can use a single line, but is it easier to read ?:`
return attributes.getTemporaryLimits().entrySet().stream().sorted(Map.Entry.comparingByKey(ACCEPTABLE_DURATION_COMPARATOR)).map(Map.Entry::getValue).map(TemporaryLimitImpl::new).collect(Collectors.toUnmodifiableList());`

I tried to sort in the store-model as you suggested, but the unit test does not work anymore (we still get a wrongly sorted collection).
I think we can keep the sort in iidm-impl, which is our interface to access the model.